File: compito36/Iper.java
package compito36;

public class Iper extends Supermercato {

    private String nome;

    //setter
    public void setNome(String nome) {
        this.nome = nome;
    }

    //getter

    public String getNome() {
        return nome;
    }

    public Iper(int codice, String indirizzo,String nome, int adde, double superf){
        super(codice,indirizzo,superf,adde);
        setNome(nome); 

    }

    public String toString() {

        return getCodice() +"\t Iper \t"+super.toString()+"\t - \t "+getSuperf()+"\t"+getAddetti()+"\t" +getNome();
    }

    
}

File: compito36_python/iper.py
from supermercato import Supermercato

class Iper(Supermercato):
    def __init__(self, codice, indirizzo, superficie, addetti, nome) -> None:
        super().__init__(codice, indirizzo, superficie, addetti)
        self.nome = nome


    def __str__(self):
        return str(self.codice) +"\t Iper \t" +super().__str__() + "\t - \t - \t" +\
            str(self.superficie)+"\t"+str(self.addetti)+"\t"+ self.nome    